src/components/products/umi/index.js

```diff
@@ -54,7 +54,6 @@ export const umi = {
             { title: 'Serializers', href: '/umi/serializers' },
             { title: 'Storage', href: '/umi/storage' },
             { title: 'Transactions', href: '/umi/transactions' },
-            { title: 'Web3Js Adapters', href: '/umi/web3js-adapters' },
```

Comment:
This will change the URI. Therefore all (external) links pointing to those, e.g. on stackoverflow will be 404.

I feel like we should not do this. It was a huge pain when we migrated from docs.metaplex.com. At least we should have a solution to forward to the new page. Not sure how to best do it from SEO perspective. @tonyboylehub ?

Reply:
https://nextjs.org/docs/pages/api-reference/next-config-js/redirects

Reply:
Probably we can "keep" the old docs up but not let them "viewable" from the index? 

If I leave there the page and delete it just from the index, it will still work from an external link. In that space we can keep the "old page" or duplicate the new page to both url

Reply:
urls should not be touched unless redirected.

We are not keeping duplicate content or pages.

Reply:
How does the redirect work?

Reply:
Multiple options. I did some testing and i feel like easiest would be to use the middleware. To do so:

1. add a middleware.js file into the `src` folder
2. add some code like this, where the object at the top has the old route and the destination.

```js
import { NextResponse,  } from 'next/server'

const redirectData = {
  "/old": {
    "destination": "/new",
    "permanent": true
  },
}

export async function middleware(request) {
  const pathname = request.nextUrl.pathname
  const redirectEntry = redirectData[pathname]

 
  if (redirectEntry ) {
    const statusCode = redirectEntry.permanent ? 308 : 307
    return NextResponse.redirect(request.nextUrl.origin + redirectEntry.destination, statusCode)
  }
 
  // No redirect found, continue without redirecting
  return NextResponse.next()
}
```

Not sure if @tonyboylehub has a different opinion how to do it.

Reply:
This worked as a wander! If @tonyboylehub doesn't have anything against it I think we should use this